TestCase/Managers/BoardManager.cs

```diff
@@ -0,0 +1,181 @@
+                    else if ((firstColElement == GlobalConsts.O_PLAYER_TOKEN))
+                    {
+                        Console.WriteLine("\nplayer O wins");
+                        return true;
+                    }
+                }
+            }
+
+            ////////////////////////////////////////////////////////////////main diagonal (TL to BR)
+
+            bool mainDiagonalMatch = true;
+            string firstMainDiagonalElement = Board[0, 0];
+
+            for (int i = 0; i < rLen; i++)
+            {
+                if (firstMainDiagonalElement != Board[i, i])
+                {
+                    mainDiagonalMatch = false;
+                    break;
+                }
+            }
+
+            if (mainDiagonalMatch)
+            {
+                if (firstMainDiagonalElement == GlobalConsts.X_PLAYER_TOKEN)
+                {
+                    Console.WriteLine("\nplayer X wins");
+                    return true;
+                }
+                else if (firstMainDiagonalElement == GlobalConsts.O_PLAYER_TOKEN)
+                {
+                    Console.WriteLine("\nplayer O wins");
+                    return true;
+                }
+            }
+
+            ////////////////////////////////////////////////////////////inverse diagonal (TR to BL)
+
+            bool inverseDiagonalMatch = true;
+            string firstInverseDiagonuleElement = Board[0, 2];
+
+            for (int i = 0; i < rLen; i++)
+            {
+                if (firstInverseDiagonuleElement != Board[i, rLen - 1 - i])
+                {
+                    inverseDiagonalMatch = false;
+                    break;
+                }
+            }
+
+            if (inverseDiagonalMatch)
+            {
+                if (firstInverseDiagonuleElement == GlobalConsts.X_PLAYER_TOKEN)
+                {
+                    Console.WriteLine("\nplayer X wins");
+                    return true;
+                }
+                else if (firstInverseDiagonuleElement == GlobalConsts.O_PLAYER_TOKEN)
+                {
+                    Console.WriteLine("\nplayer O wins");
+                    return true;
+                }
+            }
+
+            if (!inverseDiagonalMatch && !mainDiagonalMatch && noRowsMatch && noColsMatch)
+            {
+                Console.WriteLine("\nno winner yet!!!!!!");
+                return false;
+            }
+
+            return false;
+        }
+
+        public bool PlaceToken(int row, int col, string givenToken)
+        {
+            if (Board[row, col] == EMPTY_SPACE_TOKEN)
+            {
+                Board[row, col] = givenToken;
+                return true;
+            }
+
+            return false;
+        }
+
+        public void PrintBoard()
+        {
+            int rows = Board.GetLength(0);
+            int cols = Board.GetLength(1);
+
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < cols; j++)
```

Comment:
Could this 2nd for loop be simplified to the following:
`string.Join(" ", Board[i])`?